pypeit/core/flexure.py

```diff
@@ -303,147 +305,226 @@ def spec_flex_shift(obj_skyspec, arx_skyspec, arx_lines, mxshft=20):
                 corr_cen=corr.size/2, smooth=smooth_sig_pix, success=success)
 
 
-def spec_flexure_obj(specobjs, slitord, bpm, method, sky_file, mxshft=None):
-    """Correct wavelengths for flexure, object by object
+def spec_flexure_slit(slits, slitord, slitmask, bpm, sky_file, method="boxcar", specobjs=None, slitspecs=None,
+                      waveimg=None, mxshft=None, slitShift=None):
+    """Correct wavelengths for flexure, slit by slit
 
     Args:
-        specobjs (:class:`pypeit.specobjs.Specobjs`):
+        slits (:class:`pypeit.slittrace.SlitTraceSet`_):
+            Slit trace set
         slitord (`numpy.ndarray`_):
             Array of slit/order numbers
+        slitmask (`numpy.ndarray`_):
+            Slit mask
         bpm (`numpy.ndarray`_):
             True = masked slit
-        method (:obj:`str`)
-          'boxcar' -- Recommneded
-          'slitpix' --
+        method (:obj:`str`, optional)
+          'boxcar' -- Recommended for object extractions
+          'slitcen' -- Recommended when no objects are being extracted
         sky_file (str):
             Sky file
+        specobjs (:class:`pypeit.specobjs.Specobjs`_, optional):
+            Spectral extractions
+        slitspecs (list, optional):
+            A list of linetools.xspectrum1d, one for each slit
+        waveimg (`numpy.ndarray`_, optional):
+            A 2D array providing the wavelength (in A) of each pixel of an image.
+            If none, the spectral flexure correction will not be applied to the
+            wavelength image. If provided, the correction will be applied and
+            returned.
         mxshft (int, optional):
             Passed to flex_shift()
+        slitShift (list, optional):
+            A list of spectral shifts (in pixels) that have been globally applied
+            to each slit and the waveimg. This list should have the same size as
+            the number of slits.
 
     Returns:
-        list:  list of dicts containing flexure results
-            Aligned with specobjs
-            Filled with a basically empty dict if the slit is skipped or there is no object
+        tuple: Two element tuple containing (1) the input waveimg (`numpy.ndarray`_),
+               corrected for spectral flexure and (2) a list of dicts containing flexure
+               results of each slit. The latter is filled with a basically empty dict if
+               the slit is skipped
     """
     sv_fdict = None
     msgs.work("Consider doing 2 passes in flexure as in LowRedux")
+    waveimg_flex_corr = None if waveimg is None else waveimg.copy()
+
+    # Determine the method
+    slit_cen = False
+    if specobjs is None: slit_cen = True
```

Comment:
Replace with the more concise: 

slit_cen = True if specobjs is None else False